### PR TITLE
scp Is Not Parsing user@node Properly

### LINKED ISSUE #5745: Unable To Copy File Using PEM File for Automated tsh Scripts
### Description

I'm using a nightly script that copies a file from an IoT node using a PEM certificate file that has a large TTL.

**What happened**:

Copy fails with an error.

**What you expected to happen**:

The remote file should be copied over.

### Reproduction Steps
As minimally and precisely as possible, describe step-by-step how to reproduce the problem.
1. Create an SSH certificate with a large TTL:
    $ tctl auth sign --ttl=87600h --user=monitor --out=monitor.pem
2. Copy a file from a node:
   $ tsh --user monitor -i ~/work/keys/teleport/monitor.pem --proxy teleport:443 scp -d   root@node1:/tmp/var.log . 

### Server Details
- Teleport version (run `teleport version`): Teleport v5.0.0 git:v5.0.0-0-gac4971801 go1.15.5
- Server OS (e.g. from `/etc/os-release`): NAME="Ubuntu"
VERSION="20.04.1 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.1 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal

- Where are you running Teleport? (e.g. AWS, GCP, Dedicated Hardware): AWS
- Additional details:

### Client Details
- Tsh version (`tsh version`): Teleport v5.1.2 git:v5.1.2-0-g822d10b44 go1.15.5
- Computer OS (e.g. Linux, macOS, Windows): Linux
- Browser version (for UI-related issues): N/A
- Installed via (e.g. apt, yum, brew, website download): website download
- Additional details:
tsh ssh using the certificate file works fine. tsh scp without the PEM file works fine.

### Debug Logs
Please include or attach debug logs, when appropriate. Obfuscate sensitive information!
- Start Teleport with --debug flag (`teleport --debug`)
- Run tsh with --debug flag (`tsh --debug`)
INFO             Reading identity file: /home/user/work/keys/teleport/monitor.pem common/identity.go:39
DEBU [TSH]       Extracted username "monitor" from the identity file /home/user/work/keys/teleport/monitor.pem. tsh/tsh.go:1286
INFO [CLIENT]    no host login given. defaulting to user client/api.go:801
DEBU [TSH]       Pinging the proxy to fetch listening addresses for non-web ports. tsh/tsh.go:1430
DEBU [CLIENT]    not using loopback pool for remote proxy addr: teleport:443 client/api.go:2247
DEBU [CLIENT]    HTTPS client init(proxyAddr=teleport:443, insecure=false) client/weblogin.go:307
INFO [CLIENT]    Connecting to proxy to copy (recursively=false)... client/api.go:1347
INFO [CLIENT]    Connecting proxy=teleport:3023 login='user' method=0 client/api.go:1691
WARN [CLIENT]    Failed to authenticate with proxy: ssh: handshake failed: ssh: unable to authenticate, attempted methods [none publickey], no supported methods remain client/api.go:1694
INFO [CLIENT]    Connecting proxy=teleport:3023 login='user' method=1 client/api.go:1691
WARN [CLIENT]    Failed to authenticate with proxy: ssh: handshake failed: ssh: unable to authenticate, attempted methods [none publickey], no supported methods remain client/api.go:1694

ERROR REPORT:
Original Error: *trace.BadParameterError failed to authenticate with proxy teleport:3023: ssh: handshake failed: ssh: unable to authenticate, attempted methods [none publickey], no supported methods remain
Stack Trace:
        /go/src/github.com/gravitational/teleport/lib/client/api.go:1687 github.com/gravitational/teleport/lib/client.(*TeleportClient).connectToProxy
        /go/src/github.com/gravitational/teleport/lib/client/api.go:1641 github.com/gravitational/teleport/lib/client.(*TeleportClient).ConnectToProxy.func1
        /opt/go/src/runtime/asm_amd64.s:1375 runtime.goexit
User Message: failed to authenticate with proxy teleport:3023: ssh: handshake failed: ssh: unable to authenticate, attempted methods [none publickey], no supported methods remain

### Which platform(s)
- [X] Teleport Open Source
- [ ] Teleport Cloud
- [] Teleport Enterprise
- [ ] Other

### Which component(s)
- [x] Server Access
- [ ] Kubernetes Access
- [ ] Application Access
- [ ] Database Access
- [ ] CLI Clients
- [ ] Web Clients
- [x] #16311

### PR DESCRIPTION
**Pull request by thetuber**
tsh scp is not parsing "user@node" syntax properly, so that 

tsh scp ruser@node:/etc/issue /tmp

fails, but using the functionally equivalent "-l user node" syntax works:

tsh scp -l ruser node:/etc/issue /tmp

Also fixes https://github.com/gravitational/teleport/issues/5745

### CODE REVIEW

**Comment by thetuber** [comment]
https://github.com/gravitational/teleport/issues/6688

File: tool/tsh/tsh.go
Comment (awly) [inline tool/tsh/tsh.go:1570]:
```suggestion
		for _, location := range cf.CopySpec {
			// Extract username and host from "username@host:file/path"
```

**Review by awly — APPROVED** [review]

**Comment by russjones** [comment]
@awly Can you use the new buddy system when merging this in?

**Comment by awly** [comment]
Hi @thetuber,
Thanks for your contribution. We are currently unable to run our tests on
external contributions. As such, I'll be helping you validate and
merge your pull request. I've created an interal PR here: https://github.com/gravitational/teleport/pull/6927 and the CI came up clean.

Please wait for another approval (@russjones @r0mant ) and I'll get your code merged.

**Comment by russjones** [comment]
Merged in https://github.com/gravitational/teleport/pull/6927.

**Pull request closed without merging**
